### ISSUE TITLE
Issue 208: Feature Importance Analytics

### ISSUE THREAD

**Issue by ahmad1368** [ahmad1368/GrowthPilotAI#208]
📝 Description
To refine our predictive accuracy and provide actionable insights to Canadian SMBs, we must implement a system to track Feature Importance. This involves analyzing which input variables (e.g., Rent, Payroll, Utilities, Raw Materials) have the highest "weight" or "coefficient" in our regression models. By logging these importance scores, we can identify which economic factors are the primary drivers of business costs in different regions like Metro Vancouver vs. Greater Toronto Area.

 👤 User Story
As a Data Architect, I want to know which financial categories most heavily influence the "Total Monthly Spend" prediction, so I can optimize the model's feature engineering and improve the MAPE (Issue 207) in future updates.

---

 ⚙️ Technical Requirements

 1. Extracting Coefficients/Weights
 Model Interpretation: If using Linear Regression, log the normalized coefficients ($\beta$ weights). If using Tree-based models (like XGBoost/Random Forest), log the Gini Importance or Gain.
 Normalization: Scale all feature importance scores to a range of $[0, 1]$ to make comparisons across different user cohorts meaningful.

 2. Feature-Level Analytics (Anonymous Logging)
 Metadata Collection: For every successful prediction, send the "Importance Vector" to Firebase/BigQuery.
 Categories to Track:
     `fixed_costs_weight` (Rent, Insurance)
     `variable_costs_weight` (Supplies, Logistics)
     `labor_costs_weight` (Payroll, Benefits)
     `external_factors_weight` (Inflation index, Regional fuel prices)

 3. Industry Clustering
 Aggregate importance scores by industry. 
     Example: If for "Construction," the `logistics_weight` is consistently $> 0.6$, we will prioritize real-time fuel price integration in the next sprint.

---

 🛠 Feature Importance & Model Optimization Workflow

The analytics loop identifies which "lever" in the business has the most impact on the final cost prediction.

```dart
// lib/core/services/ai_optimization_service.dart

class FeatureAnalytics {
  Future<void> logFeatureWeights(Map<String, double> weights, String industry) async {
    // Weights example: {'rent': 0.15, 'payroll': 0.45, 'materials': 0.35}
    
    await FirebaseAnalytics.instance.logEvent(
      name: 'feature_importance_stats',
      parameters: {
        'industry_type': industry,
        'top_predictor': _getTopFeature(weights),
        'payroll_impact': weights['payroll'] ?? 0.0,
        'rent_impact': weights['rent'] ?? 0.0,
        'materials_impact': weights['materials'] ?? 0.0,
      },
    );
  }

  String _getTopFeature(Map<String, double> weights) {
    return weights.entries.reduce((a, b) => a.value > b.value ? a : b).key;
  }
}
```

---

 ✅ Acceptance Criteria
- [ ] The backend/edge engine calculates and extracts feature weights for every prediction.
- [ ] Data is successfully aggregated in BigQuery/Firebase without any PII.
- [ ] A monthly "Feature Heatmap" report is generated to show which costs drive inflation for SMBs in BC.
- [ ] The weights are used to re-calibrate the "Suggestion Prompts" in Issue 201 (e.g., if Payroll is the top driver, suggest "Ways to optimize labor costs").

 🔐 Security & Compliance
 Data Aggregation: We only store the relative importance of categories, not the actual dollar amounts, ensuring 100% compliance with PIPEDA.
 Algorithm Transparency: This documentation serves as a "Model Explanability" record, which is increasingly required by Canadian AI regulations to prevent biased automated decision-making.

---
## 📱 Platform Requirements
Any code written to implement this issue must be delivered across all three supported platforms:
1. **Android** — implemented for the mobile Android build.
2. **iOS** — implemented for the mobile iOS build.
3. **Web** — implemented for the web build.

**Comment by stefanoviana** [comment]
XGBoost is great for this! I'm using it in an ensemble with LightGBM for crypto trading. The key insight: XGBoost alone gets 65.5% accuracy on 4h predictions, LightGBM gets 63.2%, but the ensemble averages at 64.6% which is more stable.

https://github.com/stefanoviana/deepalpha
